go/teams/chain.go

```diff
@@ -1518,18 +1534,51 @@ func (t *TeamSigChainPlayer) addInnerLink(
 							if err != nil {
 								return err
 							}
-							if _, err := prevState.GetLatestUVWithUID(uv.Uid); err != nil {
-								return err
+							addedUIDs[uv.Uid] = true
+							_, err = prevState.GetLatestUVWithUID(uv.Uid)
+							if err == nil {
+								// Found crypto member in previous
+								// state, we are good!
+								continue
 							}
-						} else {
-							return fmt.Errorf("invalid invite type in implicit team: %v", cat)
+							_, _, found := prevState.FindActiveKeybaseInvite(uv.Uid)
+							if found {
+								// Found PUKless member in previous
+								// state, still fine!
+								continue
+							}
+							// Neither crypto member nor PUKless member
+							// found, we can't allow this addition.
+							return fmt.Errorf("Not found previous version of user %s", uv.Uid)
```

Comment:
What about when `alice,bob@rooter` becomes `alice,bob(pukless)`? Or does that not happen for some reason?

Reply:
This is what I remember is happening: When you are invited to the team as bob@rooter and prove rooter while being PUKless, your invite status will be changed to `WAITING_FOR_PUK`. This happens in `rsvp_to_team_invites`, serverside. Then, after you get a PUK, `rsvp_to_team_invites` is called again, bumping `WAITING_FOR_PUK` to `ACCEPTED`, and sending work to team_rekeyd.

I'll try it and make sure this is really what's happening.